### PR TITLE
Verify account with token

### PR DESCRIPTION
**Pull request by rafa-garcia**
CONTEXT: This changeset updates the account verification process to use non-PII parameters to get passed onto the query string.

### Workflow

- [X] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-3344
- [X] Jira ticket has the correct status.
- [X] [Changelog](CHANGELOG.md) entry added.

### Housekeeping

- [X] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers.

### CODE REVIEW

File: core/templates/core/signup.html [hunk 1]
```diff
@@ -13,12 +13,14 @@
 
     <script type="text/javascript">
 		var element = document.getElementById('sign-up');
-		var currentStep = '{% if request.GET.verify %}verification-code{% else %}credentials{% endif %}';
-		var email = '{{ request.GET.verify }}';
+		var uidb64 = '{{ request.GET.uidb64 }}';
+		var token = '{{ request.GET.token }}';
+		var currentStep = (uidb64 && token) ? 'verification-code' : 'cedentials';
```
Comment (stevejalim) [inline core/templates/core/signup.html:18]: Typo to fix?

```suggestion
		var currentStep = (uidb64 && token) ? 'verification-code' : 'credentials';
```

Comment (stevejalim) [reply core/templates/core/signup.html:18]: Not sure if the current wizard/workflow uses `cedentials` as its key

**Review by stevejalim — APPROVED** [review]
Python side LGTM (and the JS too, though that's less my thing)

File: core/templates/core/signup.html (hunk 1 above)
Comment (bobmeredith) [reply core/templates/core/signup.html:18]: Looks like the wizard currently uses 'credentials' so I'm not sure how tests are passing.
`export const STEP_CREDENTIALS = 'credentials'`

**Review by stevejalim — APPROVED** [review]

**Pull request merged**